### ISSUE TITLE
Fail with a different exit code when RuboCop encounters errors but doesn't detect violations

### ISSUE THREAD

**Issue by backus** [rubocop-hq/rubocop#3909]
So our build is currently failing because RuboCop hiccuped while analyzing our code (see #3908)

![entire build screenshot included because I'm proud](https://cloud.githubusercontent.com/assets/2085622/21974972/c88b715c-db7e-11e6-9cf6-8a1325b8992f.png)

I don't mind. Software is hard. It would be nice though if I could change the build step to something like 

```sh
bundle exec rubocop -c config/rubocop.yml
[[ $? -ne 2 ]]
```

which would only fail the build on exit code 2 which might be designated for when RuboCop violations are found but errors were not encountered.

**Comment by backus** [comment]
Reek has the following flags for this reason I think:

```sh
Exit codes:
        --success-exit-code CODE     The exit code when no smells are found (default: 0)
        --failure-exit-code CODE     The exit code when smells are found (default: 2)
```

**Comment by bbatsov** [comment]
I'm fine with adding something like this. It should be trivial to do so.

**Comment by mikegee** [comment]
The 1 and 2 might be backwards from the request here, but it looks like Rubocop already does this: https://github.com/bbatsov/rubocop/commit/5e741368

**Comment by dgollahon** [comment]
I was browsing the `good first issue` label after seeing #5795, but it looks like this issue should really be closed, seeing @mikegee's comment.

**Comment by arthurnn** [comment]
I am having this similar issue running rubocop in GitHub's action environment.
If you do a `run: bundle exec rubocop` run in an action workflow, it will fail if it returns anything besides 0 code.

It would be helpful to add a flag to the client so it would return either 0 and 1. 
 0: not config errors, or runtime errors (equivalent to 0 or 1)
 1: equivalent to 2 

If that's alright, I can work in a patch for it.
